liveblocks/liveblocks
# Fix fast LiveList.insert + LiveList.set bug

### Description

This PR ensures that insert acknowledge op is ignored if the item has been deleted by a local `LiveList.set`.

I took the opportunity to add comments and do some cleanups.

**Before merging this, we need to make sure e2e tests are passing on public and private repositories. (We might need to fix our tests setup to work with ESM)**

_Review commit by commit recommended_

#### How to test it

Upgrade our block based editor example to use `1.2.2-test3` and try to reproduce the behavior described in #731. This should also fix the "ghosting bug" mentioned by Fermat and Arcol.

#### Related issue(s)

- #628
- #731
- #1051
